paper-server/patches/sources/net/minecraft/world/level/block/entity/CreakingHeartBlockEntity.java.patch

```diff
@@ -0,0 +1,56 @@
+--- a/net/minecraft/world/level/block/entity/CreakingHeartBlockEntity.java
++++ b/net/minecraft/world/level/block/entity/CreakingHeartBlockEntity.java
+@@ -120,7 +_,7 @@
+                     Optional<Creaking> creakingProtector = creakingHeart.getCreakingProtector();
+                     if (creakingProtector.isPresent()) {
+                         Creaking creaking = creakingProtector.get();
+-                        if (!CreakingHeartBlock.isNaturalNight(level) || creakingHeart.distanceToCreaking() > 34.0 || creaking.playerIsStuckInYou()) {
++                        if (!CreakingHeartBlock.isNaturalNight(level) || creakingHeart.distanceToCreaking() > creakingHeart.getDistanceCreakingTooFar() || creaking.playerIsStuckInYou()) { // Paper - custom distance for creaking
+                             creakingHeart.removeProtector(null);
+                             return;
+                         }
+@@ -187,9 +_,10 @@
+ 
+     @Nullable
+     public static Creaking spawnProtector(ServerLevel level, CreakingHeartBlockEntity creakingHeart) {
++        if (creakingHeart.getDistanceCreakingTooFar() <= 0) return null; // Paper - avoid spawn if the max distance is 0
+         BlockPos blockPos = creakingHeart.getBlockPos();
+         Optional<Creaking> optional = SpawnUtil.trySpawnMob(
+-            EntityType.CREAKING, EntitySpawnReason.SPAWNER, level, blockPos, 5, 16, 8, SpawnUtil.Strategy.ON_TOP_OF_COLLIDER_NO_LEAVES, true
++            EntityType.CREAKING, EntitySpawnReason.SPAWNER, level, blockPos, 5, Math.min(SPAWN_RANGE_XZ, creakingHeart.getDistanceCreakingTooFar()), Math.min(SPAWN_RANGE_Y, creakingHeart.getDistanceCreakingTooFar()), SpawnUtil.Strategy.ON_TOP_OF_COLLIDER_NO_LEAVES, true // Paper - handle too far distance for avoid spawn out of limits
+         );
+         if (optional.isEmpty()) {
+             return null;
+@@ -335,6 +_,7 @@
+         } else {
+             this.clearCreakingInfo();
+         }
++        this.distanceCreakingTooFar = tag.contains(PAPER_CREAKING_MAX_DISTANCE_TAG, 99) ? tag.getInt(PAPER_CREAKING_MAX_DISTANCE_TAG) : DISTANCE_CREAKING_TOO_FAR; // Paper - Custom max distance for Creaking
```

Comment:
Use `Tag.TAG_ANY_NUMERIC` instead of 99.

Reply:
okay i base in the beacon class (then that class need use tha Tag class too) but fixed here.